src/hierarchies/likelihoods/states/base_state.h

```diff
@@ -9,25 +9,57 @@
 
 namespace State {
 
+//! Abstract base class for a generic state
+//!
+//! Given a statistical model with likelihood L(y|tau) and prior p(tau)
+//! a State class represents the value of tau at a certain MCMC iteration.
+//! In addition, each instance stores the cardinality of the number of
+//! observations in the model.
+//!
+//! State classes inheriting from this one should implement the methods
+//! `set_from_proto()` and `to_proto()`, that are used to deserialzie from
+//! (and serialize to) a `bayesmix::AlgorithmState::ClusterState`
+//! protocol buffer message.
+//!
+//! Optionally, each state can have an "unconstrained" representation,
+//! where a bijective transformation B is applied to `tau`, so that
+//! the image of B(tau) is R^d for some d.
```

Comment:
I would slightly change this sentence as follows: _so that the image of tau through B is in R^d_ or _so that its image B(tau) is in R^d, for some d_